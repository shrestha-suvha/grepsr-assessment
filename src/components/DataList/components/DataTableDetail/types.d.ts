import { IDataTableDetailList } from '../../types';

export interface IDataTableDetailProps {
  dataSetColumnDetailList: IDataTableDetailList;
}
