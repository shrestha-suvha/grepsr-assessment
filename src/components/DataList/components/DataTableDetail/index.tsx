import React, { FC } from 'react';
import { Row, Col, ProgressBar } from 'react-bootstrap';
import Table from 'react-bootstrap/Table';

import { IDataTableDetailProps } from './types';

const DataTableDetail: FC<IDataTableDetailProps> = ({ dataSetColumnDetailList }) => {
  return (
    <>
      <div className="table-wrapper border rounded-1 ">
        <Table className="fs-esm ">
          <thead className="bg-white">
            <tr>
              <th className="table-header">Column name</th>
              <th className="table-header">Description</th>
            </tr>
          </thead>
          <tbody className="">
            {(dataSetColumnDetailList || []).map(dataSetColumnDetail => (
              <tr key={dataSetColumnDetail?.id}>
                <td>
                  <Row>
                    <Col>
                      <span className="">{dataSetColumnDetail?.columnName}</span>
                      <p className="text-muted"># Integer</p>
                    </Col>
                  </Row>
                </td>
                <td>
                  <Row>
                    <Col>
                      <div style={{ color: '#1B2F50' }}>{dataSetColumnDetail?.description}</div>
                      <span className="fill-rate ">{`Fillrate: ${dataSetColumnDetail?.fillRate}%`}</span>
                      <ProgressBar now={dataSetColumnDetail?.fillRate} max={100} className="mt-2" />
                    </Col>
                  </Row>
                </td>
              </tr>
            ))}
          </tbody>
        </Table>
      </div>
    </>
  );
};
export default DataTableDetail;
