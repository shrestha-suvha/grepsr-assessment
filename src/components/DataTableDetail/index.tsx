import React, { FC } from 'react';

const DataTableDetail: FC = () => {
  return <></>;
};
export default DataTableDetail;
