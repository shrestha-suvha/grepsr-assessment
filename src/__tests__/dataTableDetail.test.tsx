import { render, screen, cleanup } from '@testing-library/react';
import { create } from 'react-test-renderer';

import DataTableDetail from 'src/components/DataList/components/DataTableDetail';

afterEach(() => {
  cleanup();
});

test('should render DataTableDetail component', () => {
  const dataSetColumnDetailList = [
    {
      columnName: 'GEO ID',
      description: 'Unique POI identifier for each locations',
      fillRate: 10,
      id: '1',
    },
  ];
  render(<DataTableDetail dataSetColumnDetailList={dataSetColumnDetailList} />);
  const dataSetColumnDetailListElement = screen.getByTestId('dataTableDetail-1');
  const tableElement = screen.getByRole('table');
  expect(dataSetColumnDetailListElement).toBeInTheDocument();
  expect(tableElement).toBeInTheDocument();
});

test('should render DataTableDetail with table header', () => {
  const dataSetColumnDetailList = [
    {
      columnName: 'GEO ID',
      description: 'Unique POI identifier for each locations',
      fillRate: 10,
      id: '1',
    },
  ];
  render(<DataTableDetail dataSetColumnDetailList={dataSetColumnDetailList} />);
  const tableHeaders = screen.getAllByRole('columnheader');
  expect(tableHeaders.length).toBe(2);
  expect(tableHeaders[0]).toHaveTextContent('Column name');
  expect(tableHeaders[1]).toHaveTextContent('Description');
});

test('should render DataTableDetail component with props value', () => {
  const dataSetColumnDetailList = [
    {
      columnName: 'GEO ID',
      description: 'Unique POI identifier for each locations',
      fillRate: 10,
      id: '1',
    },
  ];
  render(<DataTableDetail dataSetColumnDetailList={dataSetColumnDetailList} />);
  expect(screen.getAllByRole('row')[1]).toHaveTextContent('GEO ID');
  expect(screen.getAllByRole('row')[1]).toHaveTextContent('Unique POI identifier for each locations');
  expect(screen.getAllByRole('row')[1]).toHaveTextContent('10');
});

test('matches snapshot', () => {
  const tree = create(<DataTableDetail dataSetColumnDetailList={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});
