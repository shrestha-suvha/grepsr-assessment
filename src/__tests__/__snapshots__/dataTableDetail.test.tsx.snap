// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="table-wrapper border rounded-1 "
  data-testid="dataTableDetail-1"
>
  <table
    className="fs-esm  table"
  >
    <thead
      className="bg-white"
    >
      <tr>
        <th
          className="table-header"
        >
          Column name
        </th>
        <th
          className="table-header"
        >
          Description
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    />
  </table>
</div>
`;
